source/carton-runner-wasm/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::HashMap;
+
+use color_eyre::eyre::{eyre, Result};
+use wasmtime::{Engine, Store};
+use wasmtime::component::*;
+
+use carton_runner_interface::types::Tensor as CartonTensor;
+
+use crate::component::{DummyState, Model, Tensor};
+
+mod component;
+mod types;
+
+pub struct WASMModelInstance {
+    instance: Instance,
+    store: Store<DummyState>,
+    model: Model,
+}
+
+impl WASMModelInstance {
+    pub fn from_bytes(engine: &Engine, bytes: &[u8]) -> Result<Self> {
+        let comp = Component::from_binary(&engine, bytes).unwrap();
+        let mut linker = Linker::<DummyState>::new(&engine);
+        Model::add_to_linker(&mut linker, |state: &mut DummyState| state).unwrap();
+        let mut store = Store::new(&engine, DummyState);
+        let (model, instance) = Model::instantiate(&mut store, &comp, &linker).unwrap();
```

Comment:
Comments please

Reply:
Added, but I don't understand it that well, I mostly just followed the examples in the docs.